vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/JavascriptActions.java

```diff
@@ -163,7 +164,21 @@ public String getUserAgent()
     @Override
     public double getDevicePixelRatio()
     {
-        return devicePixelRatio.get();
+        double dpr;
+        if ((boolean) executeScript("return window != window.top"))
+        {
+            String href = getFrameHref();
+            webDriverProvider.get().switchTo().defaultContent();
+            dpr = devicePixelRatio.get();
+            WebElement frame = webDriverProvider.get()
+                    .findElement(By.xpath(String.format("//iframe[@src=\"%s\"]", href)));
```

Comment:
`//iframe[@src=\"%s\"]` -> `//iframe[@src='%s']`

Reply:
also, the logic looks for me as

WebdriverActions.executeInDefaultContent(Function<A,R> toRun)